tests/core/pyspec/eth2spec/test/phase_0/epoch_processing/test_process_rewards_and_penalties.py

```diff
@@ -80,6 +81,56 @@ def test_full_attestations(spec, state):
             assert state.balances[index] < pre_state.balances[index]
 
 
+@with_all_phases
+@spec_state_test
+@leaking()
+def test_full_attestations_with_leak(spec, state):
+    attestations = prepare_state_with_attestations(spec, state)
+
+    proposer_indices = [a.proposer_index for a in state.previous_epoch_attestations]
+    pre_state = state.copy()
+
+    yield from run_process_rewards_and_penalties(spec, state)
+
+    attesting_indices = spec.get_unslashed_attesting_indices(state, attestations)
+    assert len(attesting_indices) == len(pre_state.validators)
+    for index in range(len(pre_state.validators)):
+        # Proposers can still make money during a leak
+        if index in proposer_indices:
+            assert state.balances[index] > pre_state.balances[index]
+        # If not proposer but participated optimally, should have exactly neutral balance
+        elif index in attesting_indices:
+            assert state.balances[index] == pre_state.balances[index]
+        else:
+            assert state.balances[index] < pre_state.balances[index]
+
+
+@with_all_phases
+@spec_state_test
+@leaking()
+def test_partial_attestations_with_leak(spec, state):
+    attestations = prepare_state_with_attestations(spec, state)
+
+    attestations = attestations[:len(attestations) // 2]
+    state.previous_epoch_attestations = state.previous_epoch_attestations[:len(attestations)]
+    proposer_indices = [a.proposer_index for a in state.previous_epoch_attestations]
+    pre_state = state.copy()
+
+    yield from run_process_rewards_and_penalties(spec, state)
+
+    attesting_indices = spec.get_unslashed_attesting_indices(state, attestations)
+    assert len(attesting_indices) < len(pre_state.validators)
+    for index in range(len(pre_state.validators)):
+        # Proposers can still make money during a leak
+        if index in proposer_indices and index in attesting_indices:
+            assert state.balances[index] > pre_state.balances[index]
+        # If not proposer but participated optimally, should have exactly neutral balance
+        elif index in attesting_indices:
+            assert state.balances[index] == pre_state.balances[index]
+        else:
+            assert state.balances[index] < pre_state.balances[index]
```

Comment:
(not strong opinion) duplicate to `test_full_attestations_with_leak`, could be refactored.